src/gateway/environments/jsdom-sixteen/patch-against-dangling-timers.js

```diff
@@ -0,0 +1,67 @@
+// @flow
+import type {IGate, ITimerAPI} from "./types.js";
+
+/**
+ * Patch the timer API to protect against dangling timers.
+ *
+ * @returns {IGate} A gate API to control when timers should be allowed to run
+ * (gate is open), or when we should prevent them running and report dangling
+ * timers (gate is closed).
+ */
+export const patchAgainstDanglingTimers = (objToPatch: ITimerAPI): IGate => {
```

Comment:
This is mostly copied from RRS except now, rather than a flag hanging off the VM context, we use a gate interface to turn things on and off.